ClinGen/clincoded
# Connects to #530. Auth0 implementation. Auto-login support

![image](https://cloud.githubusercontent.com/assets/4326866/19537985/edaa5ec0-9606-11e6-9b50-03de8463e8f1.png)

* Switches authentication systems from Persona to Auth0
  * Supports Google Auth logins, as well as the Auth0-provided username-password database system
  * Implements Auth0 rules that triggers a Slack notification in the `#usersignups` channel when a new user sign up/logs in
  * Modifies e-mail templates for automated mail sent by Auth0
  * Adds automatic login button on test and demo instances. Pressing the `Demo Login` button should automatically log the user in to the test curator account via Auth0:
![image](https://cloud.githubusercontent.com/assets/4326866/19537978/e08f4c3c-9606-11e6-9ecb-7e5e9a078672.png)
  * Added documentation detailing the necessary setup required for Auth0, Google Auth, Slack, and local code

Testing:

1. Test normal login: login via admin and/or non-admin users on different instances
2. Test automated login: login via the `Test Login` button on supported instances (everything but `curation` and `production`)
  * Spoof HOSTS file and attempt to cross-check logins: point `curation.clinicalgenome.org` to `curation-test.clinicalgenome.org` and try the demo login, etc. Anything that points to the `curation` and `production` instances should fail the Test Login action, even if the button shows up due to address spoofing
3. Test signup and verification steps
  * Try signing up via Auth0 and/or Google, and check that all the emails and error messages are correct and functional
  * Check to see that the slack notifications trigger properly
